src/ui/networkpage.rs

```diff
@@ -287,6 +292,35 @@ impl NetworkPage {
 
         StatefulWidget::render(list, list_rect, frame.buffer_mut(), &mut self.list.state);
     }
+
+    fn render_dpc_info(&mut self, model: &Rc<Model>, rect: Rect, frame: &mut Frame) {
+        let dpc_key = model.borrow().dpc_key.clone().unwrap_or("N/A".to_string());
+
+        let configuration_string = match dpc_key.as_str() {
+            "zedagent" => "From controller".green(),
+            "manual" => "Set by local user".yellow(),
+            s => s.red(),
+        };
+
+        // convert DPC key into human readabel piece of information
+        let dpc_info = Line::default().spans(vec![
+            "Current configuration: ".white(),
+            configuration_string,
+        ]);
+
+        let mut text = Text::from(dpc_info);
+
+        if dpc_key == "manual" {
+            text.push_line(vec!["WARNING: ".red(),"the configuratiion set locally will be overwritten by working configuration from the controller".white()]);
+        }
+
+        // create paragraph with the DPC key
+        let paragraph = Paragraph::new(text)
+            .style(Style::default().fg(Color::White))
+            .alignment(Alignment::Left);
+
+        frame.render_widget(paragraph, rect);
```

Comment:
again, silly question about the approach, which is more philosophical: was there a particular reason for you to create separate render functions instead of just storing layouts, iterating them and calling render widgets?

Reply:
@uncleDecart very log explanation. Actually there is an approach you described implemented in IWidget trait. Some components follow it but I use it only for dynamic views like input field

Reply:
Okay, most probably has to do with dynamic rendering and not wanting to deep dive into optimizations, i.e. when you don't need to redraw the whole layout, got it, thanks :)